src/components/diagrams/diagram-footer.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React from 'react';
+import makeStyles from '@mui/styles/makeStyles';
+import Typography from '@mui/material/Typography';
+import AddCircleIcon from '@mui/icons-material/AddCircle';
+import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';
+import FullscreenExitIcon from '@mui/icons-material/FullscreenExit';
+import FullscreenIcon from '@mui/icons-material/Fullscreen';
+import PropTypes from 'prop-types';
+
+const useStyles = makeStyles((theme) => ({
+    counterText: {
+        bottom: theme.spacing(4),
+        left: theme.spacing(1),
+        position: 'absolute',
+    },
+    plusIcon: {
+        bottom: theme.spacing(1),
+        left: theme.spacing(4.5),
+        position: 'absolute',
+        cursor: 'pointer',
+    },
+    lessIcon: {
+        bottom: theme.spacing(1),
+        left: theme.spacing(1),
+        position: 'absolute',
+        cursor: 'pointer',
+    },
+    fullScreenIcon: {
+        bottom: theme.spacing(1),
+        right: theme.spacing(1),
+        position: 'absolute',
+        cursor: 'pointer',
+    },
+}));
+
+const DiagramFooter = (props) => {
```

Comment:
- i suggest to Move the logic of checking the props exist outside of the callback functions, this check can be moved outside of the callback function to reduce the number of times of check.

- Use the useCallback hook to memoize the callback functions, our callback functions are passed as props, they will be recreated on every render of the parent component, by using the useCallback we can prevent unnecessary re-renders.

can be something like this: 

```
    const classes = useStyles();
    const {onStopFullScreen, onStartFullScreen, onIncrementCounter, onDecrementCounter} = props;
    const handleStopFullScreen = useCallback(() => onStopFullScreen && onStopFullScreen(),[onStopFullScreen]);
    const handleStartFullScreen = useCallback(() => onStartFullScreen && onStartFullScreen(),[onStartFullScreen]);
    const handleIncrementCounter = useCallback(() => onIncrementCounter && onIncrementCounter(),[onIncrementCounter]);
    const handleDecrementCounter = useCallback(() => onDecrementCounter && onDecrementCounter(),[onDecrementCounter]);
```